onnx/common/platform_helpers.h

```diff
@@ -6,14 +6,14 @@
 
 #pragma once
 
+#include <cstdint>
+
 namespace ONNX_NAMESPACE {
 
 // Determine if the processor is little endian or not
 inline bool is_processor_little_endian() {
-  int num = 1;
-  if (*(char*)&num == 1)
-    return true;
-  return false;
+  constexpr std::int32_t value = 1;
+  return reinterpret_cast<const std::uint8_t*>(&value)[0] == 1;
```

Comment:
Wouldn't `return *(char*)&num == 1;` be more simplified? I am fine with either way tho.

Reply:
I understand they are really the same thing, but reinterpret_cast etc. is c++ syntax to make things more explicit. I can look at the disassembled code and compare

Reply:
No worries. No need to spend more time on it. Feel free to forward this PR. Thanks!